src/main/java/org/terasology/flexiblepathfinding/JPSImpl.java

```diff
@@ -70,21 +69,17 @@ public static void setStatsEnabled(boolean statsEnabled) {
      */
     public boolean run() throws InterruptedException {
         startMillis = System.currentTimeMillis();
-
-        Callable<Boolean> callable = () -> performSearch();
-        boolean result = false;
-        if (this.timeLimiter != null) {
-            try {
-                result = timeLimiter.callWithTimeout(callable, (long) (config.maxTime * 1000.0f), TimeUnit.MILLISECONDS);
-            } catch (Exception e) {
-                logger.warn("Timeout of {} exceeded: {}", config.maxTime, e.toString());
-            }
-        } else {
-            result = performSearch();
+        try {
```

Comment:
1. rxjava have some error handling. It should be using there.
2. if you left try, then recordMetrics should be in finally block (have rx some thing there?)

Reply:
went for (2), done.